src/test/kotlin/pt/up/fe/ni/website/backend/controller/AuthControllerTest.kt

```diff
@@ -0,0 +1,185 @@
+        }
+
+        @Test
+        fun `should fail when refresh token is invalid`() {
+            mockMvc.post("/auth/refresh") {
+                contentType = MediaType.APPLICATION_JSON
+                content = objectMapper.writeValueAsString(TokenDto("invalid_refresh_token"))
+            }.andExpect {
+                status { isUnauthorized() }
+                jsonPath("$.errors[0].message") { value("invalid refresh token") }
+            }
+        }
+
+        @Test
+        fun `should return new access token`() {
+            mockMvc.post("/auth/new") {
+                contentType = MediaType.APPLICATION_JSON
+                content = objectMapper.writeValueAsString(LoginDto(testAccount.email, testPassword))
+            }.andReturn().response.let { response ->
+                val refreshToken = objectMapper.readTree(response.contentAsString)["refresh_token"].asText()
+                mockMvc.post("/auth/refresh") {
+                    contentType = MediaType.APPLICATION_JSON
+                    content = objectMapper.writeValueAsString(TokenDto(refreshToken))
+                }.andExpect {
+                    status { isOk() }
+                    jsonPath("$.access_token") { exists() }
+                }
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("GET /auth/check")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class CheckToken {
+        @BeforeAll
+        fun setup() {
+            repository.save(testAccount)
+        }
+
+        @Test
+        fun `should fail when no access token is provided`() {
+            mockMvc.get("/auth").andExpect {
+                status { isForbidden() }
+                jsonPath("$.errors[0].message") { value("Access is denied") }
+            }
+        }
+
+        @Test
+        fun `should fail when access token is invalid`() {
+            mockMvc.get("/auth") {
+                header("Authorization", "Bearer invalid_access_token")
+            }.andExpect {
+                status { isUnauthorized() }
+                jsonPath("$.errors[0].message") { startsWith("An error occurred while attempting to decode the Jwt") }
+            }
+        }
+
+        @Test
+        fun `should return authenticated user`() {
+            mockMvc.post("/auth/new") {
+                contentType = MediaType.APPLICATION_JSON
+                content = objectMapper.writeValueAsString(LoginDto(testAccount.email, testPassword))
+            }.andReturn().response.let { response ->
+                val accessToken = objectMapper.readTree(response.contentAsString)["access_token"].asText()
+                mockMvc.get("/auth") {
+                    header("Authorization", "Bearer $accessToken")
+                }.andExpect {
+                    status { isOk() }
+                    jsonPath("$.authenticated_user") { value(testAccount.email) }
+                }
+            }
+        }
+
+        // TODO: Add tests for role access when implemented
```

Comment:
I don't think this TODO is necessary, tests should always be implemented in a new feature

Reply:
Right!